compaction/compaction_manager.hh

```diff
@@ -413,17 +413,17 @@ public:
 
     class compaction_reenabler {
         compaction_manager& _cm;
-        replica::table* _table;
+        compaction::table_state* _table;
```

Comment:
reference?

Reply:
The table may be removed in the background. We revalidate the pointer in submit(_table) when the compaction_reenabler is destroyed.

Reply:
submit() will bail out earlier if it finds table_state is no longer registered, which happens once underlying table is destroyed.

Reply:
BTW, a new table object can be created with the same pointer and we wouldn't be able to distinguish between them.  A UUID would provide a more robust loose coupling mechanism...

Reply:
Exactly.

But it's better to make remove() wait until there are no more instances of the table_state, rather let some stay alive and throw them away if they're invalid. The latter method is just waiting for a race to sneak in.

Reply:
just noticed we're doing it correctly.   there's `gate::holder _holder` in compaction_reenabler, which we wait for in compaction_manager::remove().

Reply:
It happens sometimes